### ISSUE TITLE
Feature Request: Enhancements to Embedded Live Preview Console Output 

### ISSUE THREAD

**Issue by gitspeaks** [microsoft/vscode-livepreview#566]
I would like to propose the following features:

1. Auto-Clear Output Window Setting: It would be beneficial to have a setting that allows users to enable or disable the automatic clearing of the output window before displaying the console.log messages from the current render instance. This feature would provide users with the flexibility to maintain a clean console environment that shows the latest render instance output without being confused by previous messages. It would be particularly helpful for those who need to track the outputs of successive render instances without the clutter of past logs.

2. Trigger Rendering and Console Output on Save Setting: Currently, the plugin triggers page rendering and console output with each source code change, which can be distracting. A configurable setting to have these actions triggered only upon saving the file would be a valuable addition. This would suit the preferences of users who find continuous rendering during code edits to be interruptive, allowing for a more focused and efficient coding experience.

Best regards,

@gitspeaks

**Comment by andreamah** [comment]
From this, I'm interpreting this as just a setting to tell the extension when to clear the output. For example, we can have the console clear on reload of the page, or to be overridden as on save?

**Comment by gitspeaks** [comment]
It’s two settings – one for clearing the last output on reload, and another for controlling when the reload occurs (e.g., on save versus every edit)

**Comment by andreamah** [comment]
For the second, do you mean something like what exists with `livePreview.autoRefreshPreview`?

**Comment by gitspeaks** [comment]
Exactly. I was not aware of this. So only a setting for clearing the output, though I prefer to have the ability to toggle this "setting" via a button (like the clear output button) in the output pane and not by editing an actual setting.

**Comment by gitspeaks** [comment]
BTW I highly recommended explicitly mentioning livePreview.autoRefreshPreview in the extension details tab description.

Currently it states:

"Live Refreshing
See the changes as you make them. By default, changes appear as you make them in the editor. You can also change this in settings to refresh the preview on save or not at all."

**Comment by kycutler** [comment]
The Live Preview extension now uses VS Code's Integrated Browser, which has native DevTools support including console. Please try it out and feel free to reopen this issue if you still see the problem there.
